app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -30,6 +30,42 @@ def supports_update?
       true
     end
 
+    def supports_operating_systems?
+      client.respond_to?(:operating_systems) || client.send(:client).respond_to?(:operating_systems)
+    end
+
+    def determine_os_type(host)
+      return nil unless host
+      ret = "other_linux"
+      return ret unless host.operatingsystem
+      base_name = host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
```

Comment:
could you extract base_name and arch_name to separate methods? especially second line seems complicated and this method is already quite long